scripts/translation/translation.py

```diff
@@ -0,0 +1,168 @@
+
+    2. **Preserve Markdown Formatting**:
+       - Ensure all Markdown formatting (e.g., `###`, `-`, `*`, `>`) and horizontal lines (`---`) remain identical to the original.
+       - IMPORTANT - Use **4 spaces for indentation** throughout the Markdown document to ensure consistent formatting.
+
+    3. **Preserve Specific Section Names in English**:
+       - Section titles such as "Tutorial Code," "Expected Output," and "Solution" must remain in English as they appear in the original file. Do not translate or alter these titles.
+
+    4. **Instructions Specific to Welcome.md File**:
+       - In the `Welcome.md` file:
+         - Translate chapter names into {language} but ensure the original chapter name is preserved in parentheses and linked correctly.
+         - Use the following Markdown format:
+           `- [Translated Name](Original%20Name)`
+           Example:
+           `- [Bonjour, le Monde!](Hello%2C%20World%21)`
+         - Translate all other text, including welcome messages and headings, into {language}.
+     - Translate to {language} all headings, as well as the chapter, and maintain the format mentioned above, like `# Welcome`, `### Coding for Kids`, `### Advanced Tutorials`, `### Getting Started`, and `### Learn the Basics` into {language}.
+     - For example, instead of 'Starting Out' it should be 'Commencer'.
+
+    5. **Do Not Translate Code or Encoded Content**:
+       - Any content enclosed in backticks (\``), code blocks, or within `<div>` tags must remain unchanged.
+       - Translate only the comments in code blocks (lines starting with `#`) into {language}, while preserving their formatting and indentation.
+
+    6. **Preserve All HTML Tags and Encoded Content**:
+       - Do not modify or translate any HTML tags or attributes, including `<div>` elements with attributes like `data-encoded`.
+
+    7. **No Additional Text or Explanations**:
+       - Do not add introductions, footnotes, or comments not present in the original content.
+       - Your response must **only contain the translated text**, adhering strictly to these instructions.
+
+    8. **Ensure Contextual Clarity**:
+       - This content is part of a programming education platform. Translations must be clear, concise, and suitable for learners, using language appropriate for technical instruction.
+
+    9. **Double-Check Critical Formatting Rules**:
+       - Verify that all headings, section names, and structures match the original format.
+       - Ensure that translated content aligns perfectly with the required structure, and untranslated sections are copied exactly.
+
+    Failure to adhere to these instructions may compromise the quality of the translation. Follow these rules strictly for every translation task.
+    """
+
+    messages = [
+        {"role": "system", "content": system_prompt},
+        {"role": "user", "content": content},
```

Comment:
you are transferring all the content instead of only what needs to be translated